### PR TITLE
Flutter screenshots stores

### PR DESCRIPTION
**Pull request by mregnauld**
How to automatically generate screenshots for a mobile Flutter app and upload them to the stores using Fastlane and Codemagic.

### CODE REVIEW

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+| --------------------------------- | ---------------------------------------- | ------------------ |
+| Android smartphone                | `1107 x 1968`                            | `3`                |
+| 7 inches Android tablet           | `1206 x 2144`                            | `2`                |
+| 10 inches Android tablet          | `1449 x 2576`                            | `2`                |
+| iPad pro 2nd gen                  | `2048 x 2732`                            | `2`                |
+| iPad pro 6th gen                  | `2048 x 2732`                            | `2`                |
+| iPhone 8 Plus                     | `1242 x 2208`                            | `3`                |
+| iPhone Xs Max                     | `1242 x 2688`                            | `3`                |
+
+Note that while the sizes for the App Store Connect have to be specifically those sizes, the Google Play Store is more permissive. Also, if you want to display what your app looks like on a tablet, prefer the portrait mode (if it still makes sense for your app, of course), so your users can see more screens on the store without any swipe.
```
Comment (maks) [inline content/knowledge-codemagic/flutter-screenshots-stores.md:92]: is it referred to as "the App Store Connect" or just "App Store Connect" ?

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+
+And for the `deployapp` command above for the App Store Connect, here is the script:
+
+{{< highlight bash "style=paraiso-dark">}}
+cd ios/
+fastlane deployapp versionName:"1.0.0" # put your own version name here
+{{< /highlight >}}
+
+Note that the App Store Connect might sometimes be buggy when it comes to deleting the previous screenshots. If that operation takes too much time (it should be done in a matter of seconds), don’t hesitate to interrupt the script and run your CI all over again.
```
Comment (maks) [inline content/knowledge-codemagic/flutter-screenshots-stores.md:401]: "run your CI" => "run your workflow"

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+title: Screenshots
+weight: 5
+aliases:
+  - /flutter/generate-upload-screenshots-stores
+  - /flutter-configuration/generate-upload-screenshots-stores
+---
+
+If you are going to publish your Flutter app to **App Store Connect** or the **Google Play Store**, you can automate the process of taking screenshots and uploading them to the stores so your users can discover your app.
+
+In order to strictly follow the steps below, you'll need to use [Riverpod](https://pub.dev/packages/flutter_riverpod) (for dependendy injection) and [intl](https://pub.dev/packages/intl) (for internationalization) in your Flutter project. While it might not be necessary for your project, it’s important to keep in mind that, if you want this approach to work, you’ll have to properly separate the UI from the logic in your code, using Riverpod or something else, so you can easily mock anything you want.
```
Comment (maks) [inline content/knowledge-codemagic/flutter-screenshots-stores.md:12]: "or something else" => "or some other means of dependency injection"

**Review by maks — CHANGES_REQUESTED** [review]
Thanks for revising, this version is much better 👍🏻 
I've just got a few language and formatting nitpicks in my review.

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+
+## Automate screenshots generation
+
+The screenshots will be generated thans to golden testing, using the [Golden Toolkit](https://pub.dev/packages/golden_toolkit) package. In the steps below, the screenshots will be illustrated as they are commonly seen in the stores. For each illustrated screenshot, here are the main steps to follow:
```
[the remark this answers is not in the document]
Comment (mregnauld) [reply content/knowledge-codemagic/flutter-screenshots-stores.md]: Fixed

File: content/knowledge-codemagic/flutter-screenshots-stores.md (hunk 1 above)
Comment (mregnauld) [reply content/knowledge-codemagic/flutter-screenshots-stores.md]: Removed the "the", though I'm not sure if I have to remove it or not.

File: content/knowledge-codemagic/flutter-screenshots-stores.md (hunk 2 above)
Comment (mregnauld) [reply content/knowledge-codemagic/flutter-screenshots-stores.md]: Changed every "CI" to "workflow"

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+
+Note that this provider can be use anywhere in your app, to fake entered text in a `TextFormField` for example.
+
+## Automatically upload screenshots to the stores
+
+To upload the screenshots to the stores, we will use the same strategy for both the Google Play Store and the App Store Connect: upload the screenshots while we deploy the app.
+
+To achieve that, we will use Fastlane. Fastlane is a set of open-source tools and scripts that automates building, testing, and deploying mobile apps for both iOS and Android.
+
+You can easily install Fastlane on your local machine with the following command: `sudo gem install fastlane`.
```
[the remark this answers is not in the document]
Comment (mregnauld) [reply content/knowledge-codemagic/flutter-screenshots-stores.md:202]: Done

File: content/knowledge-codemagic/flutter-screenshots-stores.md [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+- The `pageName` argument is the name of the image file containing your screenshot.
+- Since you'll take 2 screenshots per screen (one for the screen itself, another one for the final illustration), you’ll pass `false` for the `isFinal` argument here for the moment.
+- The `density` argument is the density of the device screen as specified above.
+- The `sizeDp` argument is the size of the device screen, where its width and height **_have to be divided by the density_**. For example, for the iPhone Xs Max, you’ll pass: `Size(1242 / 3, 2688 / 3)`.
+- The `customPump` argument, although not mandatory, can be useful in some cases. By default, the [Golden Toolkit](https://pub.dev/packages/golden_toolkit) package uses `pumpAndSettle()`, which can sometimes block the rendering if, for example, there is an infinite animation. In that case, you can pass the following argument (only for the first screenshot): `(tester) async => await tester.pump(const Duration(milliseconds: 200))`.
+
+Calling the `takeScreenshot()` function above generates an image file. You can load it into an image widget as follows:
+
+{{< highlight bash "style=paraiso-dark">}}
```
[the remark this answers is not in the document]
Comment (mregnauld) [reply content/knowledge-codemagic/flutter-screenshots-stores.md]: Fixed. I also changed other `highlight`, with `highlight text` or `highlight ruby`, so please double check to be sure.

**Comment by mregnauld** [comment]
I fixed everything as requested.
Please double check that it's ok for you, especially regarding the `highlight`.

**Review by nevercode-rudrank — APPROVED** [review]
LGTM

**Pull request merged**